use-case-guides/workday-request-time-off/README.md

```diff
@@ -208,150 +183,330 @@ curl --location 'https://{{domain}}.workday.com/ccx/api/absenceManagement/v1/{{i
+        worker_data = await fetch_worker_details(email, access_token)
+        if not worker_data:
+            raise HTTPException(status_code=404, detail="Worker not found")
+
+        worker = worker_data[0]
+        days = create_time_off_days(
+            request, worker["position"]["id"], worker["position"]["descriptor"]
+        )
+        response = await submit_time_off_request(
+            worker["workdayID"], days, access_token
+        )
+        return JSONResponse(content=response, status_code=201)
+    except httpx.HTTPError as e:
+        raise HTTPException(
+            status_code=500, detail="Failed to create time off request"
+        ) from e
+```
 
-**🗣️ Natural Language Slots**
+## Step 4: Build in Creator Studio
 
-1. Start Date: when time off starts
-2. End Date *(optional)*: when time off ends. By default end date is the same as the start date.
-3. Comments *(optional)*: Any relevant context / justification for taking time off
-4. Start Time *(optional)*: if taking partial days off, starting time for time off can be specified.
-5. End Time *(optional)*: if taking partial days off, ending time for time off can be specified.
+We will be building this plugin as a Query Resolver (fka Query-Triggered Path) along with an iPaaS to chain APIs 1, 2, and 3 together in order to submit a time off request for a user.
 
-🤲 **Inferred Slots**
+### Plugin Architecture Diagram
 
-1. Current Date: used in the "as of date" in the request PTO action. Could be static.
-2. Email Address: inferred from user roster profile of the employee calling this plugin
+![Untitled](./Plugin%20Template%20Request%20Time-Off%20in%20Workday%20081c4d522bf64bbead3697288dd46047/Untitled.png)
 
-**🗺️ Resolvers**
+**Triggers**
+
+1. Natural language input to copilot requesting to take time off
 
-1. Time Off Plan: maps an employee’s time off plans to the corresponding time off types (which is required by the request time off API)
+**Natural Language Slots**
 
-**🏃‍♂️ Actions**
+1. Start Date: when time off starts. Configured as a Date Slot.
+2. End Date *(optional)*: when time off ends. By default end date is the same as the start date if the user does not specify an end date. Configured as a Date Slot.
+3. Comments *(optional)*: Any relevant context / justification for taking time off. Configured as a Free Text Slot.
 
-1. Lookup time off type details: Executes the `/api/wql/v1/{{instance}}/data` API to query the requesting employee's workday ID, eligible time off plans, and relevant time off types through their work email address.
+**Actions**
+
+1. Lookup time off type details: Executes the `/api/wql/v1/{{instance}}/data` API to query the requesting employee's Workday ID, eligible time off plans, and relevant time off types through their work email address.
 2. Request time off: Executes the `/api/absenceManagement/v1/{{instance}}/workers/{{worker_id}}/requestTimeOff` API to request time off
 
-**📚 Guidelines**
+**Caveats**
```

Comment:
Your caveats have somewhat inconsistent parallel structure. 

1. If {{scenario}}, then {{ consequence }}
2. Keyword: {{assumption}}. {{mitigation for assumption}}
3. {{Limitation}}. {{No recovery instruction}}